### PR TITLE
Add $elemMatch return fields operator

### PR DESCRIPTION
**Pull request by magaman384**

### CODE REVIEW

**Comment by vmalloc** [comment]
It seems the build is failing on Python 2.6.

File: mongomock/collection.py [hunk 1]
```diff
@@ -354,6 +354,51 @@ def _copy_field(self, obj, container):
         else:
             return copy.copy(obj)
 
+    def _extract_projection_operators(self, fields):
+        """Removes and returns fields with projection operators."""
+        result = {}
+        allowed_projection_operators = {'$elemMatch'}
```
Comment (mdomke) [inline mongomock/collection.py]: I don't know who still uses Python 2.6, but the set syntax with curly braces is not supported there. You'll have to write `set(['$elemMatch'])`.

**Comment by vmalloc** [comment]
The Travis build is failing (I'm guessing on 2.6)...

**Comment by mdomke** [comment]
Yeah. As I pointed out this is because set literals are not available in Python 2.6. If Python 2.6 is still a version that needs to be supported, than you have to use the set constructor explicitly.

**Comment by mdomke** [comment]
@vmalloc In my opinion it would be ok to just drop support for Python 2.6. All reasonably up to date distributions ship with at least Python 2.7.x
@magaman384 Would you mind to rebase your pull request in order to resolve the merge conflicts?

**Comment by vmalloc** [comment]
@mdomke I tend to agree. I see no reason for keeping 2.6 support at this point.

**Comment by magaman384** [comment]
@mdomke No problem, done.

**Pull request merged**